### ISSUE TITLE
Conversion fails if there is a space in the file path of the program

### ISSUE THREAD

**Issue by virsagomk2** [dylanhthomas/peat-media-converter#5]
After selecting the video file (H.265 mp4) to convert, the GUI says that the video has been successfully converted, but nothing shows up inside the converter_output subfolder.

Tried converting using both preset Full Processing & Basic Processing, there's still no output inside the converter_output subfolder.

Could it be that the PEAT Media Converter only work with certain codecs like H.264 instead?

**Comment by dylanhthomas** [comment]
@virsagomk2 I haven't run into that issue before, can you post a link to a video so I can run some tests and troubleshoot?

**Comment by virsagomk2** [comment]
The video link is in this website https://mouve.au/
Snapshot attached.
![image](https://github.com/dylanhthomas/peat-media-converter/assets/23209446/54cc0a4f-5ebb-4a5a-b8d0-d8835a8dbc8e)
![image](https://github.com/dylanhthomas/peat-media-converter/assets/23209446/456244b9-f6b4-4877-9bd1-226fff7a334e)

I've also just tried re-rendered the video as h.264 mp4 & run it through the PEAT Media Converter, but there's still nothing showing up in the output folder.

Tried in both Windows 10 Intel & AMD machine, running 22H2 updates.

**Comment by dylanhthomas** [comment]
@virsagomk2 I downloaded the video from the website you mentioned and had no problem converting it or running PEAT on that output.

- Are you using the latest version 1.0.2?
- My first suggestion would be to download a fresh copy of the latest release and delete the output folder and try again.

I'm happy to jump on a quick video call sometime to troubleshoot. My email is dylan [dot] thomas [at] gmail [dot] com

**Comment by virsagomk2** [comment]
@dylanhthomas I downloaded the installer from the repo page here https://github.com/dylanhthomas/peat-media-converter/releases/download/v1.0.1/peat-media-converter-1.0.1.Setup.exe

The link in the repo still linked to v1.0.1.

When I installed it, it prompted to update, which I did & I've assumed it updated itself to v1.0.2?

Unless there's another URL to directly download the v1.0.2 version?

**Comment by dylanhthomas** [comment]
I will have to update that. The update should make you have the latest, but lots of times when I have issues with software I’ll delete and reinstall to see if that solves the issue.

https://github.com/dylanhthomas/peat-media-converter/releases/tag/v1.0.2

can we do a screen share or can you do a screen recording of the whole process?

**Comment by virsagomk2** [comment]
Screen record attached.

https://github.com/dylanhthomas/peat-media-converter/assets/23209446/46e01c2d-b044-4461-bbd3-6222c9ab9df8

**Comment by dylanhthomas** [comment]
@virsagomk2 I have some ideas, but I will need to do some testing. Thank you for your patience.

**Comment by dylanhthomas** [comment]
@virsagomk2 what I would like to do is see what error is causing this to not go through.  Rather than release a whole new build, I am going to ask you to change one line in the main file, which will tell us what is going wrong.

Open the following file in a text editor:
(your user directory)\AppData\Local\peat_media_converter\app-1.0.2\resources\app\renderer.js



Change Line 157-158 from:
```
        if (error != null)
            console.log(error)
```

to:
```
        if (error != null)
            alert(error)
```

If you rerun the tool, you should get something that looks like this:
![image](https://github.com/dylanhthomas/peat-media-converter/assets/824168/0f8abb6a-0f27-48b8-a6f7-9ba1df8c05a3)

**Comment by virsagomk2** [comment]
![image](https://github.com/dylanhthomas/peat-media-converter/assets/23209446/baf1706a-6ffe-4c28-a63a-d3b2021e5d80)

**Comment by dylanhthomas** [comment]
@virsagomk2 This helps a lot. It looks like there is an issue with a space in the path so it's not running the command.  That is something I can fix on my end. I will try to have an update posted this week.

**Comment by dylanhthomas** [comment]
@virsagomk2 I believe the 1.0.3 release fixes your issue.  You can open your app and let it download (usually takes about 10 mins to check and update), or you can download it directly: https://github.com/dylanhthomas/peat-media-converter/releases/tag/v1.0.3

Please let me know if this resolved the issue.

**Comment by virsagomk2** [comment]
Thanks @dylanhthomas but I'm still having issue, it's still not outputting the converted video.
![image](https://github.com/dylanhthomas/peat-media-converter/assets/23209446/25b79e2d-daf9-4d7d-abbe-6d5ff5f8704a)

**Comment by dylanhthomas** [comment]
Got it. That helps. Will work on it. On Oct 8, 2023, at 7:07 PM, Bryan Lee ***@***.***> wrote:﻿
Thanks @dylanhthomas but I'm still having issue, it's still not outputting the converted video.

**Comment by dylanhthomas** [comment]
@virsagomk2 I finally figured this out and fixed it in v1.0.4.

Background information:

Unless specified explicitly, nodejs executes commands using your default command shell, which depending on your Windows configuration could be either `cmd.exe` or `powershell.exe`

The commands in PEAT Media Converter were designed to run in `cmd.ex`e so for users where `powershell.exe` was the default, every conversion would fail.

This release explicitly sets the command shell to be used as `cmd.exe`
